### DISCUSSION TITLE
Font Awesome inside the TOC

### DISCUSSION THREAD

**Discussion by Dannyholt** [facebook/docusaurus#6886] [Q&A] [answered]
I'm writing an article which covers the UI of our product and I'm including Font Awesome icons in the heading like so:

```markup
### <i className="fas fa-arrows-alt"></i> Move
```

This works fine, but I noticed in the TOC, the `className` remains and so, prevents the icon from loading. Below is the markup from the TOC:

```markup
<li>
  <a href="#-move" class="table-of-contents__link toc-highlight">
    <i classname="fas fa-arrows-alt"></i> 
      Move
  </a>
</li>
```

Is this just working as intended or could we get this working? :)

**Comment by Josh-Cena** [comment, marked as answer]
It's hard to get it working. We parse each heading and make best efforts to convert them to HTML markup which then gets directly interpolated in the final output. Supporting this means hosting a whole JSX -> HTML transpiler. It's certainly desirable but the work is non-trivial.

We also want an HTML -> JSX transpiling step so users who port Markdown written for another tool can be used in MDX without any modification
